### PR TITLE
Fix description editing cursor jump

### PR DESCRIPTION
**Pull request by montehurd**
was caused by whitespace normalization on the text view being called in its `textViewDidChange`

instead just normalized whitespace on save (otherwise we'd have to dynamically adjust cursor point based on how much leading whitespace was trimmed, which seemed silly)

https://phabricator.wikimedia.org/T207646

### CODE REVIEW

File: Wikipedia/Code/DescriptionEditViewController.swift [hunk 1]
```diff
@@ -181,10 +181,12 @@ class DescriptionEditViewController: WMFScrollViewController, Themeable, UITextV
         enableProgressiveButton(false)
         wmf_hideKeyboard()
         
-        // Final trim to remove leading and trailing space
-        let descriptionToSave = descriptionTextView.text.trimmingCharacters(in: .whitespacesAndNewlines)
-        
-        guard let article = article, let dataStore = article.dataStore, let articleURL = article.url else {
+        guard
+            let descriptionToSave = descriptionTextView.normalizedWhitespaceText(),
```
Comment (joewalsh) [inline Wikipedia/Code/DescriptionEditViewController.swift]: If this is nil, do we want to show an error to the user?

Comment (montehurd) [reply Wikipedia/Code/DescriptionEditViewController.swift]: checking...

Comment (joewalsh) [reply Wikipedia/Code/DescriptionEditViewController.swift]: (nil or an empty string)

Comment (montehurd) [reply Wikipedia/Code/DescriptionEditViewController.swift]: fixing...

**Pull request merged**